File: taxicab_numbers.rb
# http://programmingpraxis.com/2012/11/09/taxicab-numbers/

# Given an arbitrary positive integer, how would you determine if it can be expressed as a sum of two cubes?

def taxicab_number?(n)
  cube_root = Math.cbrt(n).to_i
  bases = []
  (1..cube_root).each do |x|
    (1..cube_root).each do |y|
      next if x == y
      if (x**3) + (y**3) == n && !bases.include?(x||y)
        bases << x; bases << y
      end
      return true if bases.count >= 4
    end
  end
  false
end

(1..70_000).each { |n| p n if taxicab_number?(n) }

File: taxicab_numbers.js
#!/usr/bin/env node
// http://programmingpraxis.com/2012/11/09/taxicab-numbers/

function taxicabNumber(n) {
  var cubeRoot = Math.cbrt(n);
  var bases = [];
  for (var i = 1; i < cubeRoot; i++) {
    for (var j = 1; j < cubeRoot; j++) {
      if (Math.pow(i,3) + Math.pow(j,3) === n) {
        if (bases.filter(function(a) { return a != j}).length === bases.length) {
          bases.push(i); bases.push(j);
        }
      }
    }
  }
  return bases.length >= 4;
}

for (var i = 1; i < 70000; i++) {
  if (taxicabNumber(i)) { console.log(i) };
}